pyanaconda/install_tasks.py

```diff
@@ -0,0 +1,424 @@
+# installation_tasks.py
+# Container classes for running of installation tasks.
+#
+# Copyright (C) 2016  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from threading import RLock
+from pyanaconda.isignal import Signal
+import time
+import logging
+log = logging.getLogger("anaconda")
+
+
+class TaskQueue(list):
```

Comment:
I've added the BaseTask class as a parent for Task and TaskQueue to share the common code. Task queue also no longer subclasses a list, but just [emulates it](https://docs.python.org/3/reference/datamodel.html#emulating-container-types) - this has the added benefit that we can skip implementing the list methods we *don't* want to support, such as `extend()` or `reverse()` and `sort()`.